src/x/ident/tag_iterator.go

```diff
@@ -52,29 +54,69 @@ func NewTagStringsIterator(inputs ...string) (TagIterator, error) {
 
 // NewTagsIterator returns a TagsIterator over a set of tags.
 func NewTagsIterator(tags Tags) TagsIterator {
-	return newTagSliceIter(tags, nil)
+	return newTagSliceIter(tags, nil, nil)
+}
+
+// NewFieldsTagsIterator returns a TagsIterator over a set of fields.
+func NewFieldsTagsIterator(fields []doc.Field) TagsIterator {
+	return newTagSliceIter(Tags{}, fields, nil)
 }
 
 func newTagSliceIter(
 	tags Tags,
+	fields []doc.Field,
```

Comment:
This is to allow two ways of creating a tags iter? Another one from fields so we don't need to explicitly convert to tags before creating a tags iter?

Reply:
This is so we can use a single pooled type to reuse.